### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by notato8dx** [notato8dx/TowerDefense#2]
This should include documentation on code, directory structure, and branching strategy.

**Comment by notato8dx** [comment]
Code comments should explain *why* something is done.

**Comment by notato8dx** [comment]
One stance on documentation: https://groups.google.com/g/elixir-lang-talk/c/0JKd9XKmIas?pli=1

>Documentation are for folks that want to use your API without reading the source code. Code comments are for those *in the source code*.

**Comment by notato8dx** [comment]
Other opinions on documentation for private methods:
https://stackoverflow.com/questions/23145205/should-i-document-the-self-explanatory-private-methods-java
https://stackoverflow.com/questions/1743538/should-i-document-my-private-methods
https://www.reddit.com/r/PHP/comments/4fvt7m/should_i_document_private_methods/

General consensus appears to be that private code should still be documentation, but whether or not it should adhere to some standard (like JavaDoc) is up for debate, since the documentation doesn't need to be rendered outside of the source code itself.

**Comment by notato8dx** [comment]
> General consensus appears to be that private code should still be documentation, but whether or not it should adhere to some standard (like JavaDoc) is up for debate, since the documentation doesn't need to be rendered outside of the source code itself.

One important consideration is editor integration, since in VSCode it's possible to see documentation by hovering.

**Comment by notato8dx** [comment]
Thoughts on branching strategy:
- Even if a feature is not completed and ready to merge into the main branch, I'd still like to be able to play around with the changes in my other branches.
- I don't want to stop development just because I'm working on finalizing a release. So, I should be able to use a separate branch for the release finalization, while development can continue in the other section.

**Comment by notato8dx** [comment]
> Thoughts on branching strategy:
> 
> * Even if a feature is not completed and ready to merge into the main branch, I'd still like to be able to play around with the changes in my other development branches.
> * I don't want to stop development just because I'm working on finalizing a release. So, I should be able to use a separate branch for the release finalization, while development can continue in the other section.
> * Once a commit is merged into the main branch, it should be considered complete. This is because GitHub uses this as a the metric for closing issues.

Based on these thoughts, something like this might be best for the strategy:
- `master` remains untouched until a feature is completed, marking it as closed
- Feature branches are based on `master`
- Release branches are also based on `master` and only get their own patch commits. They should be re-merged into `master` once the release has been finalized, at which point a tag should be created.
- Any time `master` gets updated, all other branches should strive to merge those changes into themselves (except releases).
